check_pgactivity

```diff
@@ -5680,76 +5687,145 @@ sub check_replication_slots {
                 )
                 / (s.val)
             ),0
-        )
+        ) replslot_wal_keep,null as replslot_files
         FROM pg_replication_slots slot
         CROSS JOIN wal_size s},
+
+#  slot_name  | slot_type | replslot_wal_keep | replslot_files
+#-------------+-----------+-------------------+----------------
+# node_a_slot | physical  |                 0 |
+# slot2       | logical   |                -1 |
+
        $PG_VERSION_100 => q{
         WITH wal_size AS (
-            SELECT current_setting('wal_block_size')::int * setting::int AS val
-            FROM pg_settings
-            WHERE name = 'wal_segment_size'
+           SELECT current_setting('wal_block_size')::int * setting::int AS val
+           FROM pg_settings
+           WHERE name = 'wal_segment_size'
         )
-        SELECT slot.slot_name,
-        COALESCE(
-            floor(
-                (pg_wal_lsn_diff(
-                     CASE WHEN pg_is_in_recovery() THEN pg_last_wal_receive_lsn()
-                          ELSE pg_current_wal_lsn()
-                      END,
-		     slot.restart_lsn)
-                - (pg_walfile_name_offset(restart_lsn)).file_offset
-                )
-                / (s.val)
-            ),0
-        )
-        FROM pg_replication_slots slot
-        CROSS JOIN wal_size s}
+        SELECT slot_name,slot_type,replslot_wal_keep,
+              count(slot_file) as replslot_files -- 0 if not superuser
+        FROM
+          (SELECT slot.slot_name,slot_file,slot_type,
+          COALESCE(
+                 floor(
+                     (pg_wal_lsn_diff(pg_current_wal_lsn(), slot.restart_lsn)
+                     - (pg_walfile_name_offset(restart_lsn)).file_offset
+                     )
+                     / (s.val)
+                 ),0
+             ) as replslot_wal_keep
+            FROM pg_replication_slots slot
+            -- trick when user is not superuser
+            LEFT JOIN (
+              SELECT slot2.slot_name,
+                    pg_ls_dir('pg_replslot/'||slot2.slot_name) as slot_file
```

Comment:
Shouldn't you filter the "state" file away?
Example without the COUNT & GROUP BY:
```
          slot_name          │ slot_type │ replslot_wal_keep │          replslot_files           
─────────────────────────────┼───────────┼───────────────────┼───────────────────────────────────
 abonnement_a_publi_generale │ logical   │                 8 │ xid-5288788-lsn-16A-75000000.snap
 abonnement_a_publi_generale │ logical   │                 8 │ xid-5288788-lsn-16A-76000000.snap
 abonnement_a_publi_generale │ logical   │                 8 │ state
```

Reply:
Should be fixed now